explorer/explorer_test.go

```diff
@@ -0,0 +1,78 @@
+package explorer_test
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"go.sia.tech/core/consensus"
+	"go.sia.tech/core/types"
+	"go.sia.tech/coreutils/chain"
+	ctestutil "go.sia.tech/coreutils/testutil"
+	"go.sia.tech/explored/config"
+	"go.sia.tech/explored/explorer"
+	"go.sia.tech/explored/internal/testutil"
+	"go.sia.tech/explored/persist/sqlite"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestChainMigration(t *testing.T) {
+	log := zaptest.NewLogger(t)
+	dir := t.TempDir()
+
+	n, genesis := ctestutil.Network()
+	store, genesisState, err := chain.NewDBStore(chain.NewMemDB(), n, genesis, chain.NewZapMigrationLogger(zap.NewNop()))
+	if err != nil {
+		t.Fatal(err)
+	}
+	cm := chain.NewManager(store, genesisState)
+
+	db, err := sqlite.OpenDatabase(filepath.Join(dir, "explored.sqlite3"), log.Named("sqlite3"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer db.Close()
+
```

Comment:
Running the test like this means the tables are mostly empty. There are no transactions with siacoins, siafunds, contracts, resolutions etc. It would be good to have some basic traffic here to make sure the database is filled.
Then we corrupt it, resync and check the tip + a few of the things we added such as transactions.

Reply:
If we add contract formation/resolution code the test will end up being much larger than it currently is.  Especially because we don't have all the helpers we have in the `sqlite` package.  The genesis block has SC/SF output transactions in it at least so we could check those.